File: 프로그래머스/0/181875. 배열에서 문자열 대소문자 변환하기/배열에서 문자열 대소문자 변환하기.cpp
#include <string>
#include <vector>

using namespace std;

// a: 97, A: 65
vector<string> solution(vector<string> strArr) {
    vector<string> answer;
    for(int i=0; i<strArr.size(); i++) {
        string s = "";
        for(int j=0; j<strArr[i].size(); j++) {
            if (i%2==0) {
                if (strArr[i][j] < 97) {
                    s += strArr[i][j] + 32;
                } else s += strArr[i][j];
            } else {
                if (strArr[i][j] >= 97) {
                    s += strArr[i][j] - 32;
                } else s += strArr[i][j];
            }
        }
        answer.push_back(s);
    }
    return answer;
}

File: 프로그래머스/0/181875. 배열에서 문자열 대소문자 변환하기/배열에서 문자열 대소문자 변환하기.java
import java.util.*;
class Solution {
    public String[] solution(String[] strArr) {
        String[] answer = {};
        List<String> li = new ArrayList<>();
        for(int i=0; i<strArr.length; i++) {
            if(i%2!=0) {
                li.add(strArr[i].toUpperCase());
            } else li.add(strArr[i].toLowerCase());
        }
        answer = new String[li.size()];
        for(int i=0; i<li.size(); i++) {
            answer[i] = li.get(i);
        }
        return answer;
    }
}